src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter_Statics.cs

```diff
@@ -44,7 +45,18 @@ public static void CompileMethod(WebAssemblyCodegenCompilation compilation, WebA
             ILImporter ilImporter = null;
             try
             {
-                ilImporter = new ILImporter(compilation, method, methodIL, methodCodeNodeNeedingCode.GetMangledName(compilation.NameMangler));
+                string mangledName;
+                // TODO: We should have better detection for main or a way for native main to find out what managed main's name is
```

Comment:
Other codegens do this by using a wrapper method (see `StartupCodeMainMethod.cs`) that is `NativeCallable` and has a known symbolic name (`__managed__Main`). Codegen doesn't treat entrypoint specially.

Reply:
That sounds like a great fix once we can enable the Startup roots. Updated the comment to reflect that as the plan.